packages/api/models/files.js

```diff
@@ -96,8 +97,10 @@ class FileClass extends Manager {
     let oldFiles = (oldGranule.files || []);
 
     // all we need is the bucket and key
-    oldFiles = oldFiles.map((file) => this.getBucketAndKey(file));
-    newFiles = newFiles.map((file) => this.getBucketAndKey(file));
+    oldFiles = oldFiles.map((file) => this.getBucketAndKey(file))
```

Comment:
What was the issue here? Did files not have a bucket or key or both? I'm afraid with this change that we will end up missing files that should get deleted and maybe we need to look at why they are missing information?

Reply:
In the case of deleting a 'Failed' granule or updating a granule resulting in file changes, the files of `granule.files` may not have the `bucket,key`.  If we don't filter these {bucket: null, key: null} out, the deleting from `Files` table will fail with dynamodb `ValidationException`, and the dynamodb events will not be worked off.  I have a ticket https://bugs.earthdata.nasa.gov/browse/CUMULUS-1329